pkg/kubernetes/reconcile.go

```diff
@@ -67,64 +70,73 @@ func extract(deep interface{}) (map[string]manifest.Manifest, error) {
 	return extracted, nil
 }
 
-// walkJSON traverses deeply nested kubernetes manifest and extracts them into a flat []dict.
-func walkJSON(deep interface{}, extracted map[string]manifest.Manifest, path trace) error {
-	// array: walkJSON for each
-	if d, ok := deep.([]map[string]interface{}); ok {
-		for i, j := range d {
-			path := append(path, fmt.Sprintf("[%v]", i))
-			if err := walkJSON(j, extracted, path); err != nil {
-				return err
+// walkJSON recurses into either a map or list, returning a list of all objects that look
+// like kubernetes resources. We support resources at an arbitrary level of nesting, and
+// return an error if a node is not walkable.
+//
+// Handling the different types is quite gross, so we split this method into a generic
+// walkJSON, and then walkObj/walkList to handle the two different types of collection we
+// support.
+func walkJSON(ptr interface{}, extracted map[string]manifest.Manifest, path trace) error {
+	switch v := ptr.(type) {
+	case map[string]interface{}:
+		return walkObj(v, extracted, path)
+	case []map[string]interface{}:
+		return walkList(v, extracted, path)
+	case []interface{}:
```

Comment:
These paths in the code are new, could some tests be added to make sure `walkJSON` actually extracts proper objects from these?

https://github.com/grafana/tanka/blob/ab3a92c6ea0f8697e76ba5ebb77742a15f929311/pkg/kubernetes/reconcile_test.go#L33-L36

This one could be extended to cover not only top level arrays, but all of them